go/vt/vtgate/planbuilder/operators/offset_planning.go

```diff
@@ -38,16 +38,20 @@ func planOffsets(ctx *plancontext.PlanningContext, root Operator) Operator {
 			panic(vterrors.VT13001(fmt.Sprintf("should not see %T here", in)))
 		case offsettable:
 			newOp := op.planOffsets(ctx)
+			var anythingChanged *ApplyResult
 
 			if newOp == nil {
 				newOp = op
```

Comment:
if there is no change, we can check for reference as well, so that we can avoid this issue in future.
`newOp == nil || newOp == op`

Reply:
I refactored the code a little bit. It should be fine now.